src/raven/ext/exception.cr

```diff
@@ -1,7 +1,14 @@
+require "json"
+require "yaml"
```

Comment:
I'm lil bit wary re: these since they'll unconditionally bring the libs (or at least `yaml`, since `json` is already being used) for everyone using _raven_.

Reply:
I do see your point, there, @Sija ...

What if we added some kind of a configuration option for this? Can we access those from this file?

If so, I'm thinking something like:

```crystal
Raven.configure do |config|
  config.ignore_exception_serialization = true
end
```

Reply:
I _think_ this works?? At least, it compiles...
https://play.crystal-lang.org/#/r/ai6d

**or**, to your point, since `json` is already included, we simply modify this PR not to solve the problem for YAML as well. It does mean some inconsistency, but those could still be resolved at the application-level through the workaround listed in #83.

Reply:
I'd rather do `@type.has_constant?(:YAML)` checks.

Reply:
Something like this, @Sija?

I'm not as familiar with macro code in Crystal yet, so I want to make sure I'm thinking about it the right way:
https://play.crystal-lang.org/#/r/ai77

Reply:
Right this alley indeed, but you need to do this check at the top-level instead of the current `@type` which would be `Exception` in this case. sth along the lines of:

```cr
private YAML_LOADED = {{ @type.has_constant?(:YAML) }}

# ... and the rest of the checks is done based on this constant
# same goes for JSON
```
```

Reply:
I'm not sure though if that wouldn't be an overkill...

Reply:
I'll yield to whatever you'd prefer, @Sija! Let me know what you decide, and I'll make the change.

I agree that this might be overkill, but solving this "correctly" might be well worth it in the future when there's an obscure YAML error, we don't remember this discussion, and we end up solving the same problem twice 😄

Reply:
Ok, let's go with the current state of this PR, and revisit the case if needed.